swap/src/asb/behaviour.rs

```diff
@@ -42,6 +44,13 @@ pub enum OutEvent {
         peer: PeerId,
         error: Error,
     },
+    Registered {
+        rendezvous_node: PeerId,
+        ttl: u64,
+        namespace: Namespace,
+    },
+    RegisterFailed(RegisterError),
+    RegisterExpired(Registration),
```

Comment:
We will need our own timer for when to fire this. We might actually want to implement this as part of the rendezvous protocol itself. Even auto-renew should be possible with some config knobs.

Let's see how it pans out here.

Reply:
Given that this logic for now will be in the ASB, where do you think this logic should live?
I reckon for now we can just do this in the eventloop (out of code change simplicity), but in the long run should this logic be in the behaviour?